x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsIndex.java

```diff
@@ -118,8 +122,11 @@ private static CreateIndexRequest createIndexRequest(Clock clock, DataFrameAnaly
         String destinationIndex = config.getDest().getIndex();
         String type = mappings.keysIt().next();
         Map<String, Object> mappingsAsMap = mappings.valuesIt().next().sourceAsMap();
-        addProperties(mappingsAsMap);
-        addMetaData(mappingsAsMap, config.getId(), clock);
+        Map<String, Object> properties = getOrPutDefault(mappingsAsMap, PROPERTIES, HashMap::new);
+        checkResultsFieldIsNotPresentInSourceIndex(config, properties);
+        addProperties(properties, config, Collections.unmodifiableMap(properties));
```

Comment:
In order to avoid having to pass `properties` twice here, how about we make `addProperties` into a `Map<String, Object> createAdditionalMappings(Map<String, Object> sourceMappings, DataFrameAnalyticsConfig config)` and then we `addAll` into properties here?